### DISCUSSION TITLE
Collection cache key?

### DISCUSSION THREAD

**Discussion by meiyasan** [doctrine/doctrine2#10467] [Support Questions]
Hello, 

There are many subtleties in doctrine that I don't understand.
But why the collection cache is based on an key that involve `rootEntityName` ?

For instance how does it works in the following case: 
```
#[Entity] // <--- Some entity declaration using discrimination map
class CommonClass {}

class ExtendedClassA extends CommonClass
{
      #[ManyToOne...]
      protected $attribute;
}

class ExtendedClassB extends CommonClass
{
      #[ManyToOne...]
      protected $attribute;
}
```

The code below should be computing a key that is something like : "commonClass_XX_attribute" in both cases and so this might conflict right ? (I am looking for a second level cache issue in one application and that's how I end up here)

https://github.com/doctrine/orm/blob/ed56f42cd50a17fd00f601d57912f41769123134/lib/Doctrine/ORM/Cache/Persister/Entity/AbstractEntityPersister.php#L610

This is actually same for timestampKey, right ?
https://github.com/doctrine/orm/blob/ed56f42cd50a17fd00f601d57912f41769123134/lib/Doctrine/ORM/Cache/Persister/Entity/AbstractEntityPersister.php#L97

Thank you for your help :o)

**Comment by mpdude** [comment]
Personally, I have never used the L2 cache, so take my answer with a grain of salt.

These two PRs look related to your question:

* https://github.com/doctrine/orm/pull/1552
* https://github.com/doctrine/orm/pull/1559

In fact, the first one changed the cache key computation from the `sourceEntity` name (which would be something I'd expect) to the root entity classname, but does not further explain or mention this. If there is weak test coverage in that area, a slight change like this one may slip through and is not immediately obvious.

The second PR applied a similar-looking refactoring in another area, also switching to the root entity name without further exaplanation. It contains a test case that might help you to write a "reproducer" test yourself.

In my experience, the possibility of having to deal with entity inheritance and/or mapped superclasses is often overlooked, so you might very well have found a valid bug.

My recommendation would be to report that bug as a PR against the 2.14.x branch, with the PR adding a new, failing test. Have a look at the `tests/Doctrine/Tests/ORM/Functional/Ticket` directory; that's a good place where your new test could live.

**Reply by mpdude** [reply]
Oh, extra question: If you have one more entity class that, say, extends `ExtendedClassB`: Would you expect `ExtendedClassB` and that new class to share the collection cache, or use separate keys? Can you give reasons for your preference?

**Reply by meiyasan** [reply]
If `ExtendedClassBC` extends from `ExtendedClassB`, I expect them to share yes for a common attribute declared in B, but not based on `rootEntityName`, rather `ExtendedClassB::class`
